src/contract.rs

```diff
@@ -477,6 +515,18 @@ fn try_deposit<S: Storage, A: Api, Q: Querier>(
         ));
     }
 
+    let sender_address = deps.api.canonical_address(&env.message.sender)?;
+
+    let mut balances = Balances::from_storage(&mut deps.storage);
+    let account_balance = balances.balance(&sender_address);
+    if let Some(account_balance) = account_balance.checked_add(amount) {
+        balances.set_account_balance(&sender_address, account_balance);
+    } else {
+        return Err(StdError::generic_err(
+            "This deposit would overflow your balance",
+        ));
+    }
+
```

Comment:
in the case of depositing, moving the balance check to after the total supply check is a good optimization